src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -142,6 +143,33 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
             }
         }
 
+        private void ParseServerVariables(XElement serverVariables, UrlRewriteRuleBuilder builder)
+        {
+            if (serverVariables == null)
+            {
+                return;
+            }
+
+            foreach (var serverVariable in serverVariables.Elements(RewriteTags.Set))
+            {
+                ParseServerVariable(serverVariable, builder);
+            }
+        }
+
+        private void ParseServerVariable(XElement serverVariable, UrlRewriteRuleBuilder builder)
+        {
+            try
+            {
+                builder.AddServerVariable(
+                    serverVariable.Attribute(RewriteTags.Name).Value,
+                    _inputParser.ParseInputString(serverVariable.Attribute(RewriteTags.Value).Value));
```

Comment:
Either note or add the ability to support the Replace flag, which will replace the server variable if it already exists if true, else will throw or not replace it. To support it, simply add a check to UrlRewriteRuleBuilder to check if the list already contains a value name.